lib/cfndsl/jsonable.rb

```diff
@@ -196,6 +201,34 @@ def to_json(*args)
       as_json.to_json(*args)
     end
 
+    # rubocop:disable Metrics/PerceivedComplexity
+    def visit_json(path = '/', obj: self, visited: Set.new, &block)
```

Comment:
The existing check_refs cannot be called twice because it stores "visited" in an instance variable.  For DirectRuby usage we should be able to validate mutiple times, so I chose this approach.